src/containers/Login.js

```diff
@@ -1,81 +1,98 @@
-import '../styles/LoginSignup.css';
+import { invalidEmail } from "../helpers/stringHelper";
+
+import "../styles/LoginSignup.css";
 
 export default class Login extends React.Component {
-    constructor(props) {
-        super(props);
-        
-        this.state = {
-            email: '',
-            loginError: '',
-            password: ''
-        };
-    }
+  constructor(props) {
+    super(props);
 
-    handleChange = (key, value) => {
-        this.setState({
-            [key]: value
-        });
+    this.state = {
+      email: "",
+      loginError: "",
+      password: ""
     };
+  }
 
-    login = () => {
-        const { email, password } = this.state;
-        const request = { email, password };
+  handleChange = (key, value) => {
+    this.setState({
+      [key]: value
+    });
+  };
 
-        apiClient('POST', '/login', request)
-            .then((res) => {
-                // TODO: redirect user
-            })
-            .catch((err) => {
-                err.text().then((errorMsg) => {
-                    this.setState({
-                        loginError: errorMsg
-                    });
-                });
-            });
-    };
+  isInvalidEmail = () => {
+    const invalid = invalidEmail(this.state.email);
+
+    if (invalid) {
+      this.setState({
+        loginError:
+          "email must be less than 32 characters and contain an '@' sign"
+      });
+    }
 
-    render() {
-        return (
-            <div className='login-signup'>
-                <h1>Log In</h1>
-                <div className='login-signup-form'>
-                <input
-                    type='text'
-                    placeholder='email'
-                    value={this.state.email}
-                    onChange={e => this.handleChange('email', e.target.value)}
-                />
-                <input
-                    type='password'
-                    placeholder='password'
-                    value={this.state.password}
-                    onChange={e => this.handleChange('password', e.target.value)}
-                />
-                {this.state.loginError && (
-                    <div className='error error-show'>
-                        {this.state.loginError}
-                    </div>
-                )}
-                </div>
-                <div className='login-signup-cta'>
-                    <button className='submit-button' onClick={this.login}>
-                        login
-                    </button>
-                    <div className='login-signup-text'>
-                        No account?{" "}
-                        <Link to='/signup' className='cta-link'>
-                            Sign up
-                        </Link>
-                    </div>
-                </div>
-            </div>
-        );
+    return invalid;
+  };
+
+  login = () => {
+    const { email, password } = this.state;
+    const request = { email, password };
+
+    if (this.isInvalidEmail()) {
```

Comment:
Lmk what you think of this, but I feel like this is a little weird. isInvalidEmail is its own method and it is only being used here. The logic for setting the loginError state is also only set within isInvalidEmail. Therefore, couldn't you move that lgic here instead?

Meaning, we can just check `if (invalidEmail(this.state.email))`
and then within that block, just setState before returning rather than having a separate method that checks for an invalidEmail, changes state, and then returns a boolean

Reply:
yeah, I can see that. I left it out because if we add more validation checks the function would become unwieldy. But that's not a problem yet, so I will simplify.